Core/Systems/Affixes/Affix.cs

```diff
@@ -30,13 +32,29 @@ public void Save(TagCompound tag)
 		tag["minValue"] = MinValue;
 	}
 
-	protected void Load(TagCompound tag)
+	public void Load(TagCompound tag)
 	{
 		Value = tag.GetFloat("value");
 		MaxValue = tag.GetFloat("maxValue");
 		MinValue = tag.GetFloat("minValue");
 	}
 
+	public void NetSend(BinaryWriter writer)
+	{
+		writer.Write(GetType().FullName); // this feels like it could simply be an id instead, but for now its this.
+										  // if we used my data driven affix idea this would be so easy to make an id...
```

Comment:
If you want, you could still make a deterministic ID with any system. Make a dictionary that sorts by name and vise-versa, and then you can use string name or arbitrary ID. It's also pretty easy to do here, especially since load order is already deterministic. Not for saving/loading, but it'd work perfectly fine for networking

Reply:
Ik, but i thought we should just have something temporary that we can use for testing mp and then make something smarter later.

Reply:
> Not for saving/loading, but it'd work perfectly fine for networking

Is there a reason to not just replace the SaveData with the return of NetSend?

Reply:
It seems so similar, but IDs and names can change between reloads, assuming we eventually have dynamic or crossmod affixes. As such, you'd have to store the full name such that it can be loaded irrespective of anything else, usually in the format `ModName/AffixName`. This is how tMod handles saving modded types.

If you mean literally use the same code for both, they're just different enough to make that pointless. It'd ultimately be a wrapper around TagCompound and BitReader/Writer or something, assuming I understand your comment properly.